app/connector/mongodb/src/main/java/io/syndesis/connector/mongo/verifier/MongoConnectorVerifierExtension.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.connector.mongo.verifier;
+
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientOptions;
+import com.mongodb.MongoClientURI;
+import com.mongodb.MongoSecurityException;
+import com.mongodb.MongoSocketException;
+import io.syndesis.connector.mongo.MongoConfiguration;
+import org.apache.camel.CamelContext;
+import org.apache.camel.component.extension.verifier.DefaultComponentVerifierExtension;
+import org.apache.camel.component.extension.verifier.ResultBuilder;
+import org.apache.camel.component.extension.verifier.ResultErrorBuilder;
+import org.apache.camel.component.extension.verifier.ResultErrorHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+import static org.apache.camel.util.CastUtils.cast;
+
+public class MongoConnectorVerifierExtension extends DefaultComponentVerifierExtension {
```

Comment:
unless there is something syndesis specific, this verifier could be contributed upstream

Reply:
+1

Reply:
Was not aware of the feature in Camel, could you point me out on some example to follow up?

Reply:
I created an issue to pull that off Syndesis in a next iteration development: #6280